File: src/main/java/br/com/controllers/TelefoneController.java
package br.com.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.repository.CrudRepository;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;
import br.com.services.GenericService;
import br.com.model.entities.classes.Telefone;

@RestController
@CrossOrigin(origins = "*")
public class TelefoneController extends GenericService<Telefone, Integer> {
    private final String URL = "/telefone";

    @Autowired
    private TelefoneController(CrudRepository<Telefone, Integer> repository) {
        super(repository);
    }

    @GetMapping(value = URL)
    @ResponseBody
    private Iterable<Telefone> procurarTodos() {
        return super.findAll();
    }

    @PostMapping(value = URL)
    private Telefone salvar(@RequestBody Telefone telefone) throws Exception {
        System.out.println("\n\nTelefone a ser salvo: " + telefone.toString());
        return super.save(telefone);
    }

    @DeleteMapping(URL)
    private void remover(@RequestBody Telefone telefone) {
        System.out.println("\n\nRemovendo telefone: " + telefone.toString());
        super.remove(telefone);
        System.out.println("Telefone removido com sucesso!");
    }

    @PatchMapping(URL)
    private Telefone atualizar(@RequestBody Telefone telefone) throws Exception {
        System.out.println("\n\nAtualizando telefone" + telefone.toString());
        return super.save(telefone);
    }

    @GetMapping(URL + "/{id}")
    private Telefone procurarPorID(@PathVariable Integer id) {
        System.out.println("\n\nProcurando telefone por ID: " + id);
        var a = super.findById(id);
        System.out.println("\n\nTelefone encontrado: " + a.toString());
        return a;
    }

    @GetMapping(URL + "/Buscar/")
    private Telefone find(@RequestBody Telefone telefone) {
        return super.findById(telefone.getId());
    }

    @Override
    public void validate(Telefone entity) throws Exception {
        if (entity.getNumero().isEmpty()) {
            throw new Exception("Número não pode ser nulo");
        }
    }
}

File: src/main/java/br/com/controllers/TelefoneController.kt
package br.com.controllers

import br.com.model.Telefone
import br.com.services.TelefoneService
import org.springframework.web.bind.annotation.*

@RestController
@CrossOrigin(origins = ["*"])
class TelefoneController(private val telefoneService: TelefoneService) {
    companion object {
        private const val URL = "/telefone"
    }

    @GetMapping(value = [URL])
    @ResponseBody
    private fun procurarTodos(): Iterable<Telefone> {
        return telefoneService.findAll()
    }

    @PostMapping(value = [URL])
    @Throws(Exception::class)
    private fun salvar(@RequestBody telefone: Telefone): Telefone {
        println("\n\nTelefone a ser salvo: $telefone")
        return telefoneService.save(telefone)
    }

    @DeleteMapping(URL)
    private fun remover(@RequestBody telefone: Telefone) {
        println("\n\nRemovendo telefone: $telefone")
        telefoneService.delete(telefone)
        println("Telefone removido com sucesso!")
    }

    @PatchMapping(URL)
    @Throws(Exception::class)
    private fun atualizar(@RequestBody telefone: Telefone): Telefone {
        println("\n\nAtualizando telefone$telefone")
        return telefoneService.save(telefone)
    }

    @GetMapping("$URL/{id}")
    private fun procurarPorID(@PathVariable id: Int): Telefone {
        println("\n\nProcurando telefone por ID: $id")
        val a = telefoneService.findById(id)
        println("\n\nTelefone encontrado: $a")
        return a
    }

    @GetMapping("$URL/Buscar/")
    private fun find(@RequestBody telefone: Telefone): Telefone {
        return telefoneService.findById(telefone.id!!)
    }

    @Throws(Exception::class)
    fun validate(telefone: Telefone) {
        if (telefone.numero.isEmpty()) {
            throw Exception("Número não pode ser nulo")
        }
    }
}
